### PR TITLE
Extend CFn AWS::Lambda resources and corresponding tests

### PR DESCRIPTION
**Pull request by dominikschubert**
`aws_validated` cloudformation integration tests added for:

- [x] [AWS::Lambda::Alias](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-alias.html)
- [x] [AWS::Lambda::CodeSigningConfig](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-codesigningconfig.html)
- [x] [AWS::Lambda::EventInvokeConfig](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventinvokeconfig.html)
- [x] [AWS::Lambda::EventSourceMapping](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html)
- [x] [AWS::Lambda::Function](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html)
- [x] [AWS::Lambda::Permission](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-permission.html)
- [x] [AWS::Lambda::Url](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-url.html)
- [x] [AWS::Lambda::Version](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-version.html)

Changes besides the tests:

* Fix `AWS::Lambda::Permission`
* Fix `AWS::Lambda::Version`
* Introduce `AWS::Lambda::Alias`
* Introduce `AWS::Lambda::CodeSigningConfig`
* Fix some minor CRUD issues in the new provider

### CODE REVIEW

File: localstack/utils/cloudformation/template_deployer.py [hunk 1]
```diff
@@ -236,15 +236,16 @@ def check_not_found_exception(e, resource_type, resource, resource_status=None):
...
     ]
-    if not list(filter(lambda marker, e=e: marker in str(e), markers)):
+
+    markers_hit = [m for m in markers if m in str(e)]
+    if not markers_hit:
         LOG.warning(
             "Unexpected error retrieving details for resource type %s: Exception: %s - %s - status: %s",
             resource_type,
-            e,
+            str(e),
```
Comment (dominikschubert) [inline localstack/utils/cloudformation/template_deployer.py:245]: actually just changed this for debugging purposes but IMO it's still a bit more readable than before

**Review by dfangl — APPROVED** [review]
LGTM! just a minor found todo

File: localstack/services/cloudformation/models/awslambda.py [hunk 2]
```diff
@@ -238,47 +237,32 @@ def cloudformation_type():
...
-            result["StatementId"] = short_uid()
+        def _store_physical_id(result, resource_id, resources, resource_type):
+            parsed_statement = json.loads(result["Statement"])
+            resources[resource_id]["PhysicalResourceId"] = parsed_statement["Sid"]
+
+        def lambda_permission_params(params, resources, resource_id, **kwargs):
+            result = select_parameters("FunctionName", "Action", "Principal", "SourceArn")(
+                params, **kwargs
+            )
+            suffix = "9571XFZ"  # TODO: ?
```
[the remark this answers is not in the document]
Comment (dominikschubert) [reply localstack/services/cloudformation/models/awslambda.py]: this is now properly implemented :)

File: tests/integration/cloudformation/resources/test_lambda.py [hunk 3]
```diff
@@ -25,7 +29,9 @@ def test_lambda_autogenerated_name(cfn_client, lambda_client, deploy_cfn_templat
     assert description["StackResourceDetail"]["LogicalResourceId"] == lambda_functional_id
 
 
+# TODO: what's the purpose of this test?
```
[the remark this answers is not in the document]
Comment (dominikschubert) [reply tests/integration/cloudformation/resources/test_lambda.py]: removed the tests that didn't have a clear purpose

**Review by thrau — APPROVED** [review]
LGTM! +:100: parity

**Pull request merged**